test/pipeline/pipeline-test.cpp

```diff
@@ -0,0 +1,237 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <string>
+#include <vector>
+
+#include "src/pipeline/pipeline.hpp"
+
+#include "test/common/constants/pipeline-constants.hpp"
+#include "test/common/mocks/pipeline-mocks.hpp"
+
+namespace found {
+
+class PipelineTest : public testing::Test {
+ protected:
+    // System dependencies
+
+    // Our vector of pipeline stages
+    std::vector<std::reference_wrapper<Action>> stages;
+
+    // System under test: Pipeline
+
+    // Pipeline<Input, Output> pipeline;
+
+    void SetUp() override {
+        stages = std::vector<std::reference_wrapper<Action>>();
+        // pipeline = Pipeline<Input, Output>(stages);
+    }
+};
+
+TEST_F(PipelineTest, IntToStringToCharPipeline) {
```

Comment:
1. Make a comment above all tests. 
2. Change all comments to the following style:
```
/**
* My test case description goes here
*/
```
3. Make all test names end in `Test`. Example: `IntToStringToCharPipelineTest` instead of `IntToStringToCharPipeline`